extensions/dim-rename/src/main/java/io/druid/query/extraction/ClusterDimRenameFn.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Function;
+import com.metamx.common.StringUtils;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ *
+ */
+@JsonTypeName("rename")
+public class ClusterDimRenameFn extends AbstractDimRenameFn implements DimExtractionFn
+{
+  private static final byte CACHE_KEY = 0x07;
+
+  private final String topic;
+
+  @JsonCreator
+  public ClusterDimRenameFn(
+      @JacksonInject final RenameManager renameManager,
```

Comment:
Either way is fine, but I would like us to move in a direction where query level objects could be moved into druid-api for serde purposes and not depend on other druid modules.

My other concern would be that by leaving `@JacksonInject` parameters, it may require unnecessary binding to dummy RenameManager instances for node types that don't need to understand what those objects are, but do need to be able to deserialize / serialize or re-write part of queries. One example are the router nodes, which don't require that knowledge.